packages/issuer/lib/VcIssuer.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  encodeJsonAsURI,
+  ICredentialIssuerMetadataParametersV1_11,
+  ICredentialSuccessResponse,
+  IIssueCredentialRequest,
+  unsupported_credential_format,
+} from '@sphereon/openid4vci-common'
+import { v4 as uuidv4 } from 'uuid'
+
+export class VcIssuer {
+  _issuerMetadata: ICredentialIssuerMetadataParametersV1_11
+  _preAuthorizedCode?: string
+  _userPinRequired?: boolean
+  constructor(issuerMetadata: ICredentialIssuerMetadataParametersV1_11, preAuthorizedCode?: string, userPinRequired?: boolean) {
+    this._issuerMetadata = issuerMetadata
+    this._preAuthorizedCode = preAuthorizedCode
+    this._userPinRequired = userPinRequired
+  }
+
+  public getIssuerMetadata() {
+    return this._issuerMetadata
+  }
+
+  public createCredentialOfferDeeplink(): string {
```

Comment:
This is probably from @sksadjad 
This should really be delegated to a lower level function or class.

Reply:
yeah this is moved to another class in my [pr](https://github.com/Sphereon-Opensource/OpenID4VCI/pull/30)